internal/manager/run.go

```diff
@@ -64,6 +65,10 @@ func Run(
 	if err != nil {
 		return fmt.Errorf("failed to configure feature gates: %w", err)
 	}
+	// KongCustomEntity requires FillIDs to be enabled, because custom entities requires stable IDs to fill in its "foreign" fields.
+	if featureGates.Enabled(featuregates.KongCustomEntity) && !featureGates.Enabled(featuregates.FillIDsFeature) {
+		return fmt.Errorf("%s is required if %s is enabled", featuregates.FillIDsFeature, featuregates.KongCustomEntity)
+	}
```

Comment:
I believe placing this into `internal/manager/featuregates/feature_gates.go` will 

- make `Run()` function here not get cluttered anymore
- make other callers of `featuregates.New()` benefit from the feature gates cross validation

WDYT?

Reply:
Moved them to `featuregates.New` and added unit test cases.